tokio-trace/src/field.rs

```diff
@@ -0,0 +1,225 @@
+pub use tokio_trace_core::field::*;
+
+use std::fmt;
+use Metadata;
+
+/// Trait implemented to allow a type to be used as a field key.
+///
+/// **Note**: Although this is implemented for both the [`Field`] type *and* any
+/// type that can be borrowed as an `&str`, only `Field` allows _O_(1) access.
+/// Indexing a field with a string results in an iterative search that performs
+/// string comparisons. Thus, if possible, once the key for a field is known, it
+/// should be used whenever possible.
+pub trait AsField {
```

Comment:
Is it expected that people would implement this trait externally, or should it be sealed?